### PR TITLE
Make core.item_place_node return position of placed node.

### PR DESCRIPTION
**Pull request by BluebirdGreycoat**
I've found a need for this a few times when writing mods. Without this, mods are forced to determine the location the node was actually placed to using some guesswork. This takes away the guesswork.

Marked WIP because I might have missed something. Docs aren't updated yet. Looking for feedback.

### CODE REVIEW

**Comment by paramat** [comment]
Note, not abandoned, author is asking for input.

**Comment by DS-Minetest** [comment]
Is this really necessary? The only "guesswork" needed to do is to check before the placement whether the node at `pointed_thing.under` is `buildable_to`.
https://github.com/minetest/minetest/blob/47fbd23545044e9add069a0ec83ab525ddd056e6/builtin/game/item.lua#L296-L303

**Comment by BluebirdGreycoat** [comment]
Mods performing guesswork will most likely silently break if a future revision of this function ever extends the logic for choosing the location to put the node.

**Comment by paramat** [comment]
Removed 'WIP', that will just stop reviews. If you think it's complete then 'WIP' can be removed.

**Comment by BluebirdGreycoat** [comment]
Documentation added. I consider this finished.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -4555,7 +4555,8 @@ These functions return the leftover itemstack.
     * `param2` overrides `facedir` and wallmounted `param2`
     * `prevent_after_place`: if set to `true`, `after_place_node` is not called
       for the newly placed node to prevent a callback and placement loop
-    * returns `itemstack, success`
+    * returns `itemstack, success[, position]`
+      * `position`: the location the node was placed to. Nil if nothing was placed.
```
Comment (DS-Minetest) [inline doc/lua_api.txt]:
```suggestion
      * `position`: the location the node was placed to. `nil` if nothing was placed.
```

And I suggest to maybe remove the `[]`s around `, position`, return values are not optional.

**Comment by SmallJoker** [comment]
It would be much fancier to replace the second `success` return value with the position. If mods only checked for `if success` or `if not success` both could be combined in one. There's only a position returned if the placement succeeded.

**Comment by BluebirdGreycoat** [comment]
Updated according to suggestions.

**Comment by SmallJoker** [comment]
Quick grep through ~140 mods including large ones such as advtrains, plantlife modpack, mesecons, homedecor modpack. Results as sub-lists.

* `\.item_place_node[^_]` calls: 4, whereas 0 would be affected by this change
* `\.item_place[^_]` calls: 25, whereas 0 would be affected by this change
  * `moreblocks`: `local _, success = minetest.item_place(...)` -> no `== true` check -> compatible
  * Some mods also expect `item_place` to return `nil` value, what only may happen if `on_rightclick` returns very obscure values

We can't say for sure that it won't break any mods but the chances are low, and fixes easy. I'd rather return the position (or nil) and deal with rare mods rather than returning a redundant value in the API function.

**Comment by paramat** [comment]
@BluebirdGreycoat i see you are around recently, should this have the 'adoption needed' label?

**Comment by BluebirdGreycoat** [comment]
@paramat Yep

**Comment by sfan5** [comment]
There are no changes needed, so it doesn't need adoption either.

File: doc/lua_api.txt [hunk 2]
```diff
@@ -4555,7 +4555,8 @@ These functions return the leftover itemstack.
     * `param2` overrides `facedir` and wallmounted `param2`
     * `prevent_after_place`: if set to `true`, `after_place_node` is not called
       for the newly placed node to prevent a callback and placement loop
-    * returns `itemstack, success`
+    * returns `itemstack, position`
```
Comment (sfan5) [inline doc/lua_api.txt]: `minetest.item_place` relies on these return values, which means its docs need to be changed too

and this code too:
https://github.com/minetest/minetest/blob/8e757859d6a6bf6482480904e8485e9344e567ab/builtin/game/item.lua#L417-L418

Comment (BluebirdGreycoat) [reply doc/lua_api.txt]: Done, I think

**Comment by BluebirdGreycoat** [comment]
Just a headsup, my local git copy seems to have got into a tangle that I don't have the skills/patience to resolve at this time. If it's not suitable in the current state I think I will have to delete the entire repository and start over.

**Pull request merged**